# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineWebRTCAudio.cpp
# Commit: 20543d03a4cd
# Full Hash: 20543d03a4cda05c785011a25079aa42aec8b2ba
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-03-29 21:53:01
# Regressor Bug: 1494675
# File Overlap Count: 2
# Description:
#   Bug 1494675 - Remove AllocationHandle. r=padenot
#   
#   The handle was used to keep separate allocations of the same source in a single
#   process apart. Sources no longer use sharing, so we no longer need allocations
#   or their handles even as a concept.
# ==============================================================================

diff -r 9e7832fee028 -r 20543d03a4cd dom/media/webrtc/MediaEngineWebRTCAudio.cpp
--- a/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Wed Mar 27 14:10:27 2019 +0000
+++ b/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Wed Mar 27 14:10:24 2019 +0000
@@ -156,7 +156,6 @@
 }
 
 nsresult MediaEngineWebRTCMicrophoneSource::Reconfigure(
-    const RefPtr<AllocationHandle>&,
     const dom::MediaTrackConstraints& aConstraints,
     const MediaEnginePrefs& aPrefs, const nsString& /* aDeviceId */,
     const char** aOutBadConstraint) {
@@ -178,7 +177,7 @@
     GetErrorName(rv, name);
     LOG("Mic source %p Reconfigure() failed unexpectedly. rv=%s", this,
         name.Data());
-    Stop(nullptr);
+    Stop();
     return NS_ERROR_UNEXPECTED;
   }
 
@@ -393,12 +392,8 @@
 nsresult MediaEngineWebRTCMicrophoneSource::Allocate(
     const dom::MediaTrackConstraints& aConstraints,
     const MediaEnginePrefs& aPrefs, const nsString& aDeviceId,
-    const ipc::PrincipalInfo& aPrincipalInfo, AllocationHandle** aOutHandle,
-    const char** aOutBadConstraint) {
+    const ipc::PrincipalInfo& aPrincipalInfo, const char** aOutBadConstraint) {
   AssertIsOnOwningThread();
-  MOZ_ASSERT(aOutHandle);
-
-  *aOutHandle = nullptr;
 
   mState = kAllocated;
 
@@ -424,8 +419,7 @@
   return rv;
 }
 
-nsresult MediaEngineWebRTCMicrophoneSource::Deallocate(
-    const RefPtr<const AllocationHandle>&) {
+nsresult MediaEngineWebRTCMicrophoneSource::Deallocate() {
   AssertIsOnOwningThread();
 
   MOZ_ASSERT(mState == kStopped || mState == kAllocated);
@@ -485,7 +479,6 @@
 }
 
 void MediaEngineWebRTCMicrophoneSource::SetTrack(
-    const RefPtr<const AllocationHandle>&,
     const RefPtr<SourceMediaStream>& aStream, TrackID aTrackID,
     const PrincipalHandle& aPrincipal) {
   AssertIsOnOwningThread();
@@ -547,8 +540,7 @@
   StartStop mAction;
 };
 
-nsresult MediaEngineWebRTCMicrophoneSource::Start(
-    const RefPtr<const AllocationHandle>&) {
+nsresult MediaEngineWebRTCMicrophoneSource::Start() {
   AssertIsOnOwningThread();
 
   // This spans setting both the enabled state and mState.
@@ -587,8 +579,7 @@
   return NS_OK;
 }
 
-nsresult MediaEngineWebRTCMicrophoneSource::Stop(
-    const RefPtr<const AllocationHandle>&) {
+nsresult MediaEngineWebRTCMicrophoneSource::Stop() {
   AssertIsOnOwningThread();
 
   LOG("Mic source %p Stop()", this);
@@ -655,12 +646,12 @@
   AssertIsOnOwningThread();
 
   if (mState == kStarted) {
-    Stop(nullptr);
+    Stop();
     MOZ_ASSERT(mState == kStopped);
   }
 
   MOZ_ASSERT(mState == kAllocated || mState == kStopped);
-  Deallocate(nullptr);
+  Deallocate();
   MOZ_ASSERT(mState == kReleased);
 }
 
@@ -1160,27 +1151,23 @@
 }
 
 void MediaEngineWebRTCAudioCaptureSource::SetTrack(
-    const RefPtr<const AllocationHandle>&,
     const RefPtr<SourceMediaStream>& aStream, TrackID aTrackID,
     const PrincipalHandle& aPrincipalHandle) {
   AssertIsOnOwningThread();
   // Nothing to do here. aStream is a placeholder dummy and not exposed.
 }
 
-nsresult MediaEngineWebRTCAudioCaptureSource::Start(
-    const RefPtr<const AllocationHandle>&) {
+nsresult MediaEngineWebRTCAudioCaptureSource::Start() {
   AssertIsOnOwningThread();
   return NS_OK;
 }
 
-nsresult MediaEngineWebRTCAudioCaptureSource::Stop(
-    const RefPtr<const AllocationHandle>&) {
+nsresult MediaEngineWebRTCAudioCaptureSource::Stop() {
   AssertIsOnOwningThread();
   return NS_OK;
 }
 
 nsresult MediaEngineWebRTCAudioCaptureSource::Reconfigure(
-    const RefPtr<AllocationHandle>&,
     const dom::MediaTrackConstraints& aConstraints,
     const MediaEnginePrefs& aPrefs, const nsString& aDeviceId,
     const char** aOutBadConstraint) {